src/emulator/auth/state.ts

```diff
@@ -922,6 +907,19 @@ interface TemporaryProofRecord {
   // a bit easier. Therefore, there's no need to record createdAt timestamps.
 }
 
+export function encodeRefreshToken(refreshTokenRecord: RefreshTokenRecord): string {
+  return Buffer.from(JSON.stringify(refreshTokenRecord), "utf8").toString("base64");
+}
+
+export function decodeRefreshToken(refreshTokenString: string): RefreshTokenRecord {
+  try {
+    const json = Buffer.from(refreshTokenString, "base64").toString("utf8");
+    return JSON.parse(json) as RefreshTokenRecord;
```

Comment:
Let's validate the object has the `_AuthEmulatorRefreshToken` field to prevent misuse of other types of tokens, e.g. custom tokens, fake IDP tokens, mfa pending credentials

Reply:
Cool, missed that assertion, thanks!